module/VuFind/src/VuFind/Search/Base/Params.php

```diff
@@ -970,20 +970,24 @@ public function addCheckboxFacet($filter, $desc)
      * @param string $default Default field name (null for default behavior).
      *
      * @return string         Human-readable description of field.
-     *
-     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function getFacetLabel($field, $value = null, $default = null)
     {
+        $checkboxFacet = $this->checkboxFacets[$field]["$field:$value"] ?? null;
```

Comment:
It seems like an oversight that facet aliases are not applied for checkbox facets. Perhaps that's better addressed in a separate PR, but I wonder if we should resolve the alias first, and then check for checkboxes (we'd have to change the behavior of `addCheckboxFilter()` as well to account for aliasing and keep things consistent -- obviously it would be wise to add coverage for weird edge cases to the test suite).

Reply:
Makes sense. I changed the order so that aliases are checked first and checkbox facets before normal facets. Fortunately checkbox facets are added with `addFilter()` (there's no `addCheckboxFilter()`, right?), so that should already work.

Reply:
...and yes, on reviewing the code, I think it would be a good idea to make `addCheckboxFacet` alias-aware to ensure consistent behavior. Would you like to handle that here, or should I address it in a separate PR?

Reply:
Now I must admit that I'm not sure what it means to make `addCheckboxFacet` alias-aware. Since it's called with values from configuration, what does it need to handle?

Reply:
The configuration could still use an alias as the field name. I think we need to map those to "true" names in order for things to work consistently. We could also potentially log a warning if we want to discourage that behavior.

Reply:
You wouldn't do it intentionally, but it could happen by accident and cause subtle problems... that's why I suggest making it work but logging a warning.

Reply:
Ok. How about addFacet then? Shouldn't it have the same mapping logic then?

Reply:
Yes, I'd think they should be consistent. I probably assumed it already did. :-)

Reply:
Nope. :) And then there's at least `getFacetOperator` as well. Sorry, I'm still not convinced this is the right path. If I miss any of the places where the mapping should have been added as well, the result is much worse than what we have now.

Reply:
That's fine, it's a narrow edge case and at this point should be addressed separately if at all since the scope is larger than I thought.

Reply:
Ok. Was there something else that needs to be addressed here in this PR?

Reply:
No, I don't think so, unless you think it's worth adding an addFacetAlias method (which I doubt -- as noted here, that might be better done as a subsequent PR, if at all).

Reply:
A subsequent PR, if anything, is fine for me. :)